plugins/plugins.gradle.kts

```diff
@@ -10,17 +10,18 @@ subprojects {
         }
         val sourceJar = task<Jar>("sourceJar") {
             description = "Creates a JAR that contains the source code."
-            from(java.sourceSets["main"].allSource)
+            from(project.sourceSets["main"].allSource)
             classifier = "sources"
         }
+        val javaPluginConvention = convention.plugins["java"] as JavaPluginConvention
         val javadocJar = task<Jar>("javadocJar") {
             dependsOn("javadoc")
             description = "Creates a JAR that contains the javadocs."
-            from(java.docsDir)
+            from(javaPluginConvention.docsDir)
```

Comment:
Out of curiosity, can you just use `from(tasks.named("javadoc")` here?

Reply:
Yep!